gcp_variant_transforms/beam_io/vcf_parser.py

```diff
@@ -259,17 +264,40 @@ def __init__(self,
     self._file_name = file_name
     self._allow_malformed_records = allow_malformed_records
 
-    text_source = textio._TextSource(
-        file_pattern,
-        0,  # min_bundle_size
-        compression_type,
-        True,  # strip_trailing_newlines
-        coders.StrUtf8Coder(),  # coder
-        validate=False,
-        header_processor_fns=(
-            lambda x: not x.strip() or x.startswith('#'),
-            self._process_header_lines),
-        **kwargs)
+    if splittable_bgzf:
+      text_source = BGZFBlockSource(
+          file_name,
+          range_tracker,
+          representative_header_lines,
+          compression_type,
+          header_processor_fns=(
+              lambda x: not x.strip() or x.startswith('#'),
+              self._process_header_lines),
+          **kwargs)
+    elif compression_type == filesystems.CompressionTypes.GZIP:
+      text_source = BGZFSource(
```

Comment:
The build in Text one does not support BGZF, i.e., the reading would fail. Given one file that ends with .gz without index file present, we cannot easily tell whether it is non-blocked gzip or blocked gzip. The BGZFSource extended the build-in class, and it is able to read both BGZF and GZIP file. There is no performance degradation for reading GZIP using BGZFSource.